wallet/wallet-cli-lib/src/commands/mod.rs

```diff
@@ -51,368 +51,507 @@ use self::helper_types::{
+    #[clap(name = "token-mint")]
     MintTokens {
+        /// The token id of the tokens to be minted
         token_id: String,
+        /// The receiving address of the minted tokens
         address: String,
+        /// The amount to be minted
         amount: String,
     },
 
     /// Unmint existing tokens and reduce the total supply
+    /// Unminting reduces the total supply and puts the unminted tokens back at the issuer's control.
+    /// The wallet must own the tokens that are being unminted.
+    #[clap(name = "token-unmint")]
     UnmintTokens {
+        /// The token id of the tokens to be unminted
         token_id: String,
+        /// The amount to be unminted
         amount: String,
     },
 
-    /// Lock the circulating supply for the token
+    /// Lock the circulating supply for the token. THIS IS IRREVERSIBLE.
+    /// Tokens that can be locked will lose the ability to mint/unmint them
+    #[clap(name = "token-lock-supply")]
     LockTokenSupply {
+        /// The token id of the token, whose supply will be locked
         token_id: String,
     },
 
-    /// Freezing the token forbids any operation with all the tokens (except for optional unfreeze)
+    /// Freezing the token (by token authority) forbids any operation with all the tokens (except for the optional unfreeze).
+    ///
+    /// After a token is frozen, no transfers, spends, or any other operation can be done.
+    /// This wallet must own the authority keys to be able to freeze.
+    #[clap(name = "token-freeze")]
     FreezeToken {
+        /// The token id of the token to be frozen.
         token_id: String,
+        /// Whether these tokens can be unfrozen again, or permanently freeze them.
         is_unfreezable: CliIsUnfreezable,
     },
 
-    /// By unfreezing the token all operations are available for the tokens again
+    /// By unfreezing the token all operations are available for the tokens again.
+    ///
+    /// Notice that this is only possible if the tokens were made to be unfreezable during freezing.
+    /// This wallet must own the authority keys to be able to unfreeze.
+    #[clap(name = "token-unfreeze")]
     UnfreezeToken {
+        /// The token id of the token to be unfrozen.
         token_id: String,
     },
 
-    /// Change the authority of a token
-    ChangeTokenAuthority {
+    /// Send a given token amount to a given address. The wallet will automatically calculate the required information
+    #[clap(name = "token-send")]
+    SendTokensToAddress {
+        /// The token id of the tokens to be sent
         token_id: String,
+        /// The destination address receiving the tokens
         address: String,
+        /// The amount of tokens to be sent
+        amount: String,
     },
 
-    /// Rescan
-    Rescan,
-
-    SyncWallet,
-
+    /// Get the total balance in this wallet. See available options to include more categories, like locked coins.
+    #[clap(name = "address-balance")]
```

Comment:
I am not convinced by the `address-` prefix. If I am not mistaken, it adds up the balance of all addresses in the account. What's the idea here? Maybe `account-` works better? It could also be applied to the utxo listing command.

Reply:
How about something stupid: `subwallet-`? :laughing: Trying to come up with a name for the set of addresses generated by fixing a particular account index in the derivation path but too many things are already called `account`...

Reply:
Consider that for usability "subwallet" doesn't really mean anything... I would rather make it just "balance" as an exception.

Reply:
I will keep this for now and maybe we can change it later if we find something better. Let's see whether people can suggest something better.

Reply:
> Trying to come up with a name for the set of addresses generated by fixing a particular account index in the derivation path but too many things are already called `account`...

I see there already is `account-create` and `account-select` where `account-` is used in this way. From that point of view, `account-balance` and `account-utxos` sound pretty sensible. However, there is still an ambiguity with "account" used for staking. Or do we just use the word address (perhaps with some modifier/prefix) for that?

Reply:
I can live with `account-balance`, and cast out staking. 

Besides the `wallet-` commands, almost all commands have to do with the "selected account". This is why it's difficult to include staking in `account-`. The account is selected during the whole session. So `account-` doesn't really mean that the command has to do with the current account, but it means something like "account control", where accounts can be switched.